File: C++/temperature/temperature.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
// Initialise la temperature minimun
int tempMin = 0;
int main()
{
    int n; // the number of temperatures to analyse
    cin >> n; cin.ignore();
    for (int i = 0; i < n; i++) {
        int t; // a temperature expressed as an integer ranging from -273 to 5526
        cin >> t; cin.ignore();
        if (tempMin == 0){
            tempMin = t;
        }
        else if (abs(t) < abs(tempMin)){
                tempMin = t;
            }
        else if (abs(t) == abs(tempMin)){
                tempMin = abs(t);
            }
        }
    
    cout << tempMin << endl;
}

File: Python3/Temperature/temperature.py
import sys
import math

tempsMin=0
n = int(input())  # the number of temperatures to analyse
# Parcours toutes les températures
for i in input().split():
    
    # t: a temperature expressed as an integer ranging from -273 to 5
    t = int(i)

    #Gere exception si aucune temperature
    if n == 0:
        tempsMin=0
    elif tempsMin == 0:
        tempsMin = t
    elif abs(tempsMin) > abs(t):
        tempsMin = t
    elif abs(tempsMin) == abs(t):
        if t > tempsMin:
            tempsMin = t
            
# Affiche le résultat
print(tempsMin)
